pkg/provider/cloud/aks/provider.go

```diff
@@ -235,14 +245,18 @@ func ConvertStatus(provisioningState string, powerState armcontainerservice.Code
 	}
 }
 
-func ConvertMDStatus(provisioningState string, powerState armcontainerservice.Code) apiv2.ExternalClusterMDState {
+func ConvertMDStatus(provisioningState string, powerState armcontainerservice.Code, allNodesRunning bool) apiv2.ExternalClusterMDState {
 	switch {
 	case provisioningState == string(resources.CreatingAKSMDState):
 		return apiv2.ProvisioningExternalClusterMDState
 	case provisioningState == string(resources.SucceededAKSMDState) && string(powerState) == string(resources.RunningAKSMDState):
 		return apiv2.RunningExternalClusterMDState
-	case provisioningState == string(resources.FailedAKSMDState):
+	case provisioningState == string(resources.SucceededAKSMDState) && string(powerState) == string(resources.StoppedAKSMDState):
+		return apiv2.StoppedExternalClusterMDState
+	case provisioningState == string(resources.FailedAKSMDState) && !allNodesRunning:
```

Comment:
if will think about this condition:
this basically says, if last operation failed and readyreplicas != replicas, then md state = failed,
suppose if last Upgrade operation fails, and hence readyreplicas != replicas , but still some nodes are ready, should we declare MD as Failed/Unusable

maybe when provisioningState == string(resources.FailedAKSMDState) and ready replicas == 0, then we can declare md as unusable
wdyt?